### PR TITLE
ceph: Fix missing patch for Ceph with dmcrypt

### LINKED ISSUE #334975: ceph 18.2.4 still broken with cryptsetup/dmcrypt
### Describe the bug
The fix rolled out in #334275 lacks the third patch (https://github.com/ceph/ceph/commit/607eb34b2c278566c386efcbf3018629cf08ccfd) described in #334227. Without that patch it is still not possible to use ceph-volume with a dmcrypt'd volume.

### Steps To Reproduce
1. `ceph-volume lvm activate` on a bluestore OSD using LUKS

### Expected behavior
Volume is activated.

### Additional context
As [described in the original issue thread](https://github.com/NixOS/nixpkgs/issues/334227#issuecomment-2285317255) there is still a stack trace that can be fixed by applying the above (and in the other issue thread) mentioned patch.

### Notify maintainers

- @nh2
- @adevress @alexanderkjeldaas @johanot @krav

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
% nix run nixpkgs#nix-info -- -m 
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.45, NixOS, 24.05 (Uakari), 24.05pre-git`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.90.0`
 - nixpkgs: `/nix/store/yj1w6dachan720i1m4037jllrf2r5xrb-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by nh2**
## Description of changes

Fixes #334975.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by nh2** [comment]
CC @benaryorg. It was already tested in that issue, so we can merge if it builds.

**Comment by benaryorg** [comment]
Any chance we can get this merged and backported soon-ish?
*nixos-24.05* has been broken for Ceph dmcrypt users for about three weeks at this point.

**Comment by nh2** [comment]
I rebased, re-tested, and added the NixOS VM test contributed in PR #336580 that covers this. Passes for me. Merging.

Closes #336580.

**Comment by nh2** [comment]
> Any chance we can get this merged and backported soon-ish?

Working on that now.

**Comment by nh2** [comment]
> Working on that now.

@benaryorg Please review the backport: #338630

File: nixos/tests/ceph-single-node-bluestore-dmcrypt.nix
Comment (dotlambda) [inline nixos/tests/ceph-single-node-bluestore-dmcrypt.nix:1]: Why was this merged with a failing check? This new file should have been formatted with nixfmt.

Comment (nh2) [reply nixos/tests/ceph-single-node-bluestore-dmcrypt.nix:1]: @dotlambda It was a direct cherry-pick from https://github.com/NixOS/nixpkgs/pull/336580/checks where all checks passed. I'm not sure why the check didn't run there.

Should I make a PR to fix it?

Comment (nh2) [reply nixos/tests/ceph-single-node-bluestore-dmcrypt.nix:1]: Ah, I see. It was only the maintainer list change

```diff
-      maintainers = [ benaryorg nh2 ];
+      maintainers = [
+        benaryorg
+        nh2
+      ];
```

that was accidental and already removed again in https://github.com/NixOS/nixpkgs/pull/338619/files

**Pull request merged**